scrapy/dupefilters.py

```diff
@@ -59,10 +58,10 @@ def request_seen(self, request):
             return True
         self.fingerprints.add(fp)
         if self.file:
-            self.file.write(fp.hex() + '\n')
+            self.file.write(fp + '\n')
 
     def request_fingerprint(self, request):
-        return self.fingerprinter.fingerprint(request)
+        return self.fingerprinter.fingerprint(request).hex()
```

Comment:
That's unfortunate we won't be able to get efficiency gains from switching to bytes (less memory usage). So we're making a step in a right direction, but not actually fixing the issue. 

We would still need to make a backwards-incompatible change to RFPDupeFilter (or figure out a workaround) to actually make it. I think that's fine to handle this separately, if we're sure the current code changes allow us to do that easier. What do you think @Gallaecio?

Reply:
I agree with changing it separately.

Given the duplicate filter class API (which made sense before centralized fingerprings), the only backward-compatible way forward I can think of would be to deprecate the class in favor of a new one that uses binary fingerprints.